pkg/operator/ceph/server.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2020 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package operator
+
+import (
+	"github.com/pkg/errors"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+type AdmissionController struct {
+	providerName string
+	certDir      string
+	port         int
+	scheme       *runtime.Scheme
+}
+
+func NewAdmissionController(providerName string, port int, certDir string, scheme *runtime.Scheme) *AdmissionController {
+	return &AdmissionController{
+		providerName: providerName,
+		port:         port,
+		certDir:      certDir,
+		scheme:       scheme,
+	}
+}
+
+var resources = []webhook.Validator{&cephv1.CephCluster{}, &cephv1.CephBlockPool{}}
+
+// StartServer will start the server
+func (a *AdmissionController) StartServer() error {
+	logger.Infof("starting the webhook for backend %q", a.providerName)
+
+	opts := ctrl.Options{
```

Comment:
For example, related to my other comment, can we just initialize the scheme, port, and certDir here?

Reply:
I've moved the inits to server.go